src/components/App/App.js

```diff
@@ -40,6 +43,33 @@ class App extends Component {
     })
   }
 
+  setFavorites = (newFavorite) => {
+    const allIDs = this.state.favorites.reduce((acc, favorite) => {
+      acc.push(favorite.id);
+      return acc;
+    }, []);
+    if (!allIDs.includes(newFavorite.id)) {
+      this.saveFavorite(newFavorite);
+    } else {
+      this.removeFromSaved(newFavorite);
+    }
+  };
```

Comment:
Checks all IDs of favorites in state to see if the new favorite is already in there. If it is, the favorite it removed. If it is not, the favorite is added to state.